File: LayoutTests/http/tests/misc/xhtml.php
<html>
<body>
<script>
if (window.testRunner)
    testRunner.dumpAsText();

var accept = "<?php echo $_SERVER["HTTP_ACCEPT"]; ?>";
document.write("Accept: " + accept + "<br><br>");
if (accept.match(/application\/xhtml\+xml/)) {
    document.write("PASS: The browser asks for XHTML.");
} else {
    document.write("FAIL: The browser doesn't ask for XHTML");
}
</script>

</body>
</html>

File: LayoutTests/http/tests/misc/xhtml.py
#!/usr/bin/env python3

import os
import sys

accept = os.environ.get('HTTP_ACCEPT', '')

sys.stdout.write(
    'Content-Type: text/html\r\n\r\n'
    '<html>\n'
    '<body>\n'
    '<script>\n'
    'if (window.testRunner)\n'
    '    testRunner.dumpAsText();\n'
    '\n'
    'var accept = "{}";\n'
    'document.write("Accept: " + accept + "<br><br>");\n'
    'if (accept.match(/application\\/xhtml\\+xml/)) {{\n'
    '    document.write("PASS: The browser asks for XHTML.");\n'
    '}} else {{\n'
    '    document.write("FAIL: The browser doesn\'t ask for XHTML");\n'
    '}}\n'
    '</script>\n'
    '\n'
    '</body>\n'
    '</html>\n'.format(accept)
)